opm/autodiff/LinearisedBlackoilResidual.hpp

```diff
@@ -63,6 +63,8 @@ namespace Opm
         /// pressure specification.
         ADB well_eq;
 
+        std::array<double, 3> matbalscale;
```

Comment:
For more generality, I think this should be an std::vector<double> rather than an std::array. The constructor will give it the correct size, but it would then be possible for other models to override it (polymer for example) in the future (it would have no effect on the current code).